packages/suite-desktop-core/src/modules/firmware.ts

```diff
@@ -0,0 +1,91 @@
+import { FirmwareType } from '@trezor/connect';
+import { buildFirmwareFileName } from '@trezor/connect/src/utils/firmwareUtils';
+import { DeviceModelInternal, VersionArray } from '@trezor/device-utils';
+
+import { readDir, saveBuffer } from '../libs/user-data';
+import { app } from '../typed-electron';
+
+import type { ModuleInit } from './index';
+
+export const SERVICE_NAME = '@trezor/firmware';
+
+const FIRMWARE_DIR = '/firmware';
+
+export const getStoredFirmwares = async () => {
+    const userDataDir = app?.getPath('userData');
+    const resp = await readDir(FIRMWARE_DIR);
+    if (!resp.success) {
+        return { success: false, error: resp.error };
+    }
+    const { payload } = resp;
+
+    return {
+        success: true,
+        payload: {
+            firmwareDir: `${userDataDir}${FIRMWARE_DIR}/`,
+            firmwareList: payload,
+        },
+    };
+};
+
+export const init: ModuleInit = ({ mainThreadEmitter }) => {
```

Comment:
The whole `firmware` module is currently unused, right? Only `getStoredFirmwares` util is imported somewhere. Therefore, `saveBuffer` util is in my opinion unused as well.

Reply:
Yes, it will be used in next PR where I am adding the FW downloaded and event `FIRMWARE_DOWNLOADED` will be emitted by connect. Just separating it here so PRs are smaller to review.